### ISSUE TITLE
Failures during "make test" with default stable Python abi on Fedora 41

### ISSUE THREAD

**Issue by zdohnal** [vim/vim#15534]
### Steps to reproduce

1. have Fedora 41 (I thought it can be Python 3.13 related, but its version is the same as with passing build - but maybe it really can play its part)
2. configure and run tests under normal user - ```./configure --enable-gui=no --enable-python3interp=dynamic && make test```

Ends with failures - 
[fail.txt](https://github.com/user-attachments/files/16678778/fail.txt)

I'll try to look into once I return from vacation and I'm filing the issue if someone else hits it.

### Expected behaviour

The tests pass.

### Version of Vim

9.1.0682

### Environment

Fedora 41
TERM=xterm-256color

EDIT:

Since stable ABI is no longer default, the configure command has to be ```./configure --enable-gui=no --enable-python3interp=dynamic && make test```

**Comment by zdohnal** [comment]
Looks some other update solved it - I was not able to find out which one :( .

**Comment by k-takata** [comment]
Now, Stable ABI is disabled by default again.
Does the test pass if Stable ABI is enabled explicitly?

**Comment by zdohnal** [comment]
@k-takata ouch, yeah, the reason why the test started to pass was because the stable abi was disabled by default, but it reintroduced https://github.com/vim/vim/issues/15457 , which got fixed by https://github.com/ychin/vim/commit/e2e0d0a2c1026f01ef6a1a28261aa088a0c7e963 for unstable ABI.

Either way, stable and unstable API fails in test suite with Python 3.13 now :( .

**Comment by zdohnal** [comment]
I've discovered one issue in Vim with errmsg severity and error printing in Vim - see the mentioned PR.

There is one more issue remaining after applying the patch:

```
Failures:
	Found errors in Test_python3_iter_ref():
	command line..script /home/user/vim/src/testdir/runtest.vim[607]..function RunTheTest[57]..Test_python3_iter_ref line 38: Expected 1 but got 0
```

I don't know how to debug Python code when it is run from Vim (I use PuDB for python debugging and it did not get me anywhere), so I didn't get far, but I see difference in reference count when I ran part of the code via Python 3.12 and 3.13. sys.getrefcount() with 3.12 returns sane-looking numbers (4 or 5), whereas 3.13 returns 4294967295, which is UINT_MAX or -1 incorrectly assigned into uint variable.

The Vim commands I used on Vim command line are:

```
:py3 import sys; import vim; v = vim.buffers; base_count = sys.getrefcount(v); print(base_count)
```

Python 3.13
```
4294967295
```

Python 3.12
```
5
```
